qiskit_ibm_runtime/execution_span.py

```diff
@@ -0,0 +1,94 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Execution span classes."""
+
+from typing import Union, Iterable, Sequence, TypeVar
+from datetime import datetime
+from dataclasses import dataclass
+
+
+"""The format accepted by ``numpy.ndarray.__getitem__()``."""
+SliceType = tuple[Union[slice, int, list[int]], ...]
```

Comment:
I don't see a situation where we'll have something other than `slice` in the data slices. So I think we can remove the definition of `SliceType` here and below replace `SliceType` with `slice`. @ihincks Was there a specific reason to define `SliceType`?

Reply:
In fact, it is possible for a batch job to have two different slices, each corresponding to the same pub result. Therefore I think we should go for a list of integers.